### DISCUSSION TITLE
cartopy doesn't seem to be working with pyart

### DISCUSSION THREAD

**Discussion by salbenson** [ARM-DOE/pyart#1376] [Q&A]
Hi, 
I am trying to use pyart.  I seem to get so many warnings from cartopy that the pyart plot won't complete.  I have tried updating cartopy with conda update cartopy, but it doesn't help.  I try updating pyart with conda update -c conda-forge arm_pyart but that doesn't help either.  I get a weird display and an empty image file.  
Any help is appreciated.
Thanks!
Sally

You are using the Python ARM Radar Toolkit (Py-ART), an open source
library for working with weather radar data. Py-ART is partly
supported by the U.S. Department of Energy as part of the Atmospheric
Radiation Measurement (ARM) Climate Research Facility, an Office of
Science user facility.

If you use this software to prepare a publication, please cite:
    JJ Helmus and SM Collis, JORS 2016, doi: 10.5334/jors.119

/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/mpl/geoaxes.py:1608: MatplotlibDeprecationWarning: This usage of Quadmesh is deprecated: Parameters meshWidth and meshHeights will be removed; coordinates must be 2D; all parameters except coordinates will be keyword-only.
  antialiased=antialiased, shading=shading, **kwargs)
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/mpl/gridliner.py:307: UserWarning: The .xlabels_top attribute is deprecated. Please use .top_labels to toggle visibility instead.
  warnings.warn('The .xlabels_top attribute is deprecated. Please '
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/mpl/gridliner.py:343: UserWarning: The .ylabels_right attribute is deprecated. Please use .right_labels to toggle visibility instead.
  warnings.warn('The .ylabels_right attribute is deprecated. Please '
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/mpl/geoaxes.py:1608: MatplotlibDeprecationWarning: This usage of Quadmesh is deprecated: Parameters meshWidth and meshHeights will be removed; coordinates must be 2D; all parameters except coordinates will be keyword-only.
  antialiased=antialiased, shading=shading, **kwargs)
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/crs.py:245: ShapelyDeprecationWarning: __len__ for multi-part geometries is deprecated and will be removed in Shapely 2.0. Check the length of the `geoms` property instead to get the  number of parts of a multi-part geometry.
  if len(multi_line_string) > 1:
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/crs.py:297: ShapelyDeprecationWarning: Iteration over multi-part geometries is deprecated and will be removed in Shapely 2.0. Use the `geoms` property to access the constituent parts of a multi-part geometry.
  for line in multi_line_string:
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/crs.py:364: ShapelyDeprecationWarning: __len__ for multi-part geometries is deprecated and will be removed in Shapely 2.0. Check the length of the `geoms` property instead to get the  number of parts of a multi-part geometry.
  if len(p_mline) > 0:
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/io/__init__.py:260: DownloadWarning: Downloading: https://naciscdn.org/naturalearth/110m/physical/ne_110m_coastline.zip
  warnings.warn('Downloading: {}'.format(url), DownloadWarning)
/Users/u0079358/miniconda3/lib/python3.7/site-packages/cartopy/io/__init__.py:260: DownloadWarning: Downloading: https://naciscdn.org/naturalearth/110m/cultural/ne_110m_admin_1_states_provinces_lines.zip
  warnings.warn('Downloading: {}'.format(url), DownloadWarning)
(base) mace-mac-07:python u0079358$

**Comment by mgrover1** [comment]
@salbenson - can you try updating from Python 3.7 to 3.10?

```bash
conda install python=3.10
```

In the meantime, we can take a look at troubleshooting.

**Reply by salbenson** [reply]
@mgrover1  - Hi Max, thanks for your help.  I tried updating python, but it found package conflicts.  I suspect it is some sort of conflict between matplotlib, cartopy and pyart.  I don't know how to resolve python package conflicts.  Do I have to completely uninstall python?   

**Here is the conflict message**:
The following specifications were found to be incompatible with your system:
  - feature:/osx-64::__osx==10.16=0
  - feature:|@/osx-64::__osx==10.16=0
  - arm_pyart -> matplotlib-base -> __osx[version='>=10.11']
  - cartopy -> matplotlib-base[version='>=1.5.1'] -> __osx[version='>=10.11']
  - matplotlib -> matplotlib-base[version='>=3.6.2,<3.6.3.0a0'] -> __osx[version='>=10.11']
  - matplotlib-base -> __osx[version='>=10.11']

**Here is the beginning output from the python update.**  
salbenson$ conda install python=3.10
Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                     /  

UnsatisfiableError: The following specifications were found to be incompatible with a past
explicit spec that is not an explicit spec in this operation (python):

  - python=3.10

The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

**Reply by mgrover1** [reply]
I would try this first:

```bash
conda update --all
```

This should update everything in your environment, helping to resolve those issues. After that, you can try running 

```bash
conda install python>3.9
```

**Comment by salbenson** [comment]
I haven't been able to update python yet.  I tried "conda update --all" and "conda install python=3.9" and "conda install python=3.10", but it finds conflicting packages so it remains at python 3.7.12.  It may be because I initially installed miniconda3 on my Mac and not anaconda.  At this point, to update my Python version, I'll have to contact my computer support team and get back to you when that has been completed.
